cfme/common/__init__.py

```diff
@@ -89,3 +90,209 @@ def remove_tag(self, tag):
+                value = tuple(tds[1:])
+                try:
+                    rowspan = int(self._sel.get_attribute(key, 'rowspan'))
+                except (ValueError, TypeError):
+                    rowspan = None
+                if rowspan:
+                    key_values.append((key, key_id, [value]))
+                else:
+                    key_values.append((key, key_id, value))
+            else:
+                # value of last key_values should be extended
+                key_values[-1][2].append(tuple(tds))
+        for key, key_id, value in key_values:
+            value_object = process_field(self._sel, value)
+            setattr(self, key_id, value_object)
+            self._keys.append(key_id)
+
+    def reload(self):
+        self._object.load_details(refresh=True)
+        return self.load()
+
+
+def process_field(sel, values):
+    if isinstance(values, list):
+        return map(partial(process_field, sel), values)
+    else:
+        if len(values) == 1:
+            return process_single_value(sel, values[0])
+        else:
+            return map(partial(process_single_value, sel), values)
+
+
+def process_single_value(sel, single_value):
+    text_value = sel.text_sane(single_value)
+    if len(text_value) == 0:
+        value = None
+    else:
+        # Try parsing a number
+        try:
+            value = int(text_value)
+        except ValueError:
+            try:
+                value = float(text_value)
+            except ValueError:
+                value = text_value
+    if sel.get_attribute(sel.element('..', root=single_value), 'onclick'):
+        link = single_value
+    else:
+        link = None
+    img = None
+    try:
+        img_o = sel.element('./img', root=single_value)
+        img = urlparse(sel.get_attribute(img_o, 'src').strip())
+    except sel.NoSuchElementException:
+        pass
+
+    if isinstance(value, float):
+        return extfloat(sel, value, link, img)
+    elif isinstance(value, int):
+        return extint(sel, value, link, img)
+    elif isinstance(value, str):
+        return extstr(sel, value, link, img)
+    else:
+        return value
+
+
+class extmixin(object):
+    def _custom_click_handler(self, wait_ajax):
+        if self.link is None:
+            raise ValueError("Cannot click on {} because it is not clickable".format(repr(self)))
+        try:
+            return self.sel.click(self.link, wait_ajax, no_custom_handler=True)
+        except self.sel.StaleElementReferenceException:
+            raise RuntimeError('Couldnt click on {} because the page was left.'.format(repr(self)))
+
+    @property
+    def clickable(self):
+        return self.link is not None
+
+    def click(self):
+        """A convenience function to click the summary item."""
+        if self.clickable:
+            self.sel.click(self)
+        else:
+            raise RuntimeError('Cannot click {} because it is not clickable'.format(repr(self)))
+
+
+class extfloat(float, extmixin):
```

Comment:
People will probably want to do `obj.summary.asd.fgh.click()`, `obj.summary.aasd.fgh.img.path` and similar things, they will not want to go around it using things like `obj.click_on("Asd", "Fgh")` or `obj.get_image_url("asdasfas", "werq")`. That is why I decided to do with a bit of injection here. The extent is quite limited, only to this mixin. I also don't like subclassing builtins, but this seems to be not as evil as it could be.